### PR TITLE
Fix #8

### LINKED ISSUE #8: Imageio in test_requires causes enum34 and futures to be installed
This is the weirdest issue I've come across in a while. I have a package that is installed in a conda environment. Tests for this package are run with `python setup.py test` which runs tests from the `test_suite` keyword argument in `setup()`. My problem is if I have `imageio` 2.4+ installed from conda-forge and `imageio` listed in `test_requires`, it tries to install `enum34` and `futures` packages in a Python 3.6 environment *both* of which should not be needed in this environment. The `imageio` library is the only thing causing this issue and *only* when installed from conda/conda-forge. I see this issue locally on OSX, travis OSX and linux, and appveyor.

Here is a test script that you can see the results with by running `python setup.py test`. The `setup.py` file:

```
from setuptools import setup, find_packages
setup(name='mypkg',
      version='0.1',
      test_suite='mypkg.tests.suite',
      packages=find_packages(),
      zip_safe=False,
      install_requires=[],
      tests_require=['imageio'],
      )
```

I originally thought the issue was related to new versions of pip and setuptools but the issue still exists after downgrading.

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# packages in environment at /Users/davidh/anaconda/envs/polar2grid_py36:
#
# Name                    Version                   Build  Channel
affine                    2.2.1                      py_0    conda-forge
aggdraw                   1.3.6a0.dev0              <pip>
alabaster                 0.7.12                     py_0    conda-forge
appdirs                   1.4.3                      py_1    conda-forge
appnope                   0.1.0                 py36_1000    conda-forge
arrow                     0.12.1                py36_1002    conda-forge
asn1crypto                0.24.0                py36_1003    conda-forge
atomicwrites              1.2.1                      py_0    conda-forge
attrs                     18.2.0                     py_0    conda-forge
babel                     2.6.0                      py_1    conda-forge
backcall                  0.1.0                      py_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.5                        py_1    conda-forge
beautifulsoup4            4.6.3                 py36_1000    conda-forge
behave                    1.2.6                    py36_0    conda-forge
binaryornot               0.4.4                      py_1    conda-forge
blas                      1.1                    openblas    conda-forge
bleach                    3.0.2                      py_0    conda-forge
bokeh                     1.0.1                 py36_1000    conda-forge
boost                     1.67.0           py36h3e44d54_0    conda-forge
boost-cpp                 1.67.0               h3a22d5f_0    conda-forge
boto3                     1.9.38                     py_0    conda-forge
botocore                  1.12.39                    py_0    conda-forge
bottleneck                1.2.1            py36h7eb728f_1    conda-forge
bs4                       0.0.1                     <pip>
bumpversion               0.5.3                     <pip>
bzip2                     1.0.6                         1    conda-forge
ca-certificates           2018.10.15           ha4d7672_0    conda-forge
cachey                    0.1.1                     <pip>
cairo                     1.14.12              he6fea26_5    conda-forge
cartopy                   0.16.0           py36h81b52dc_2    conda-forge
certifi                   2018.10.15            py36_1000    conda-forge
cffi                      1.11.5           py36h5e8e0c9_1    conda-forge
cftime                    1.0.2.1          py36h7eb728f_0    conda-forge
chardet                   3.0.4                 py36_1003    conda-forge
clangdev                  7.0.0                h94d2629_0    conda-forge
click                     7.0                        py_0    conda-forge
click-plugins             1.0.4                      py_0    conda-forge
cligj                     0.5.0                      py_0    conda-forge
cloudpickle               0.6.1                      py_0    conda-forge
collective.checkdocs      0.2                       <pip>
commonmark                0.7.5                      py_0    conda-forge
configobj                 5.0.6                      py_0    conda-forge
cookiecutter              1.6.0                 py36_1000    conda-forge
coverage                  4.5.1            py36h470a237_1    conda-forge
cryptography              2.3.1            py36hdffb7b8_0    conda-forge
cryptography-vectors      2.3.1                 py36_1000    conda-forge
curl                      7.62.0               h74213dd_0    conda-forge
cycler                    0.10.0                     py_1    conda-forge
cython                    0.29             py36hfc679d8_0    conda-forge
cytoolz                   0.9.0.1          py36h470a237_1    conda-forge
dask                      0.20.0                     py_0    conda-forge
dask-core                 0.20.0                     py_0    conda-forge
dcw-gmt                   1.1.4                         0    conda-forge
decorator                 4.3.0                      py_0    conda-forge
distributed               1.24.0                py36_1000    conda-forge
docutils                  0.14                  py36_1001    conda-forge
donfig                    0+untagged.14.g6f19083           <pip>
eccodes                   2.8.2                ha8b302a_0    conda-forge
ecmwf_grib                1.27.0               h0cee55e_1    conda-forge
entrypoints               0.2.3                 py36_1002    conda-forge
expat                     2.2.5                hfc679d8_2    conda-forge
fftw                      3.3.8                h470a237_0    conda-forge
fiona                     1.7.13           py36hb00a9d7_4    conda-forge
flake8                    3.6.0                 py36_1000    conda-forge
fontconfig                2.13.1               hce039c3_0    conda-forge
freetype                  2.9.1                h6debe1e_4    conda-forge
freexl                    1.0.5                h470a237_2    conda-forge
future                    0.17.0                py36_1000    conda-forge
gdal                      2.2.4            py36hb00a9d7_9    conda-forge
geocoder                  1.38.1                     py_0    conda-forge
geos                      3.6.2                hfc679d8_3    conda-forge
geotiff                   1.4.2                h700e5ad_5    conda-forge
geoxarray                 0.post7+g2b6c9dc           <pip>
gettext                   0.19.8.1             h1f1d5ed_1    conda-forge
ghostscript               9.22                 hfc679d8_1    conda-forge
giflib                    5.1.4                h470a237_1    conda-forge
gitchangelog              3.0.3                     <pip>
glib                      2.55.0               h464dc38_2    conda-forge
gmt-python                0+unknown                 <pip>
graphviz                  2.38.0               hd4f1ed2_9    conda-forge
gshhg-gmt                 2.3.7                         0    conda-forge
h5netcdf                  0.6.2                      py_0    conda-forge
h5pickle                  0.3                       <pip>
h5py                      2.8.0            py36h7eb728f_3    conda-forge
hdf4                      4.2.13               h951d187_2    conda-forge
hdf5                      1.10.2               hc401514_2    conda-forge
heapdict                  1.0.0                 py36_1000    conda-forge
holoviews                 1.10.9                     py_1    conda-forge
html5lib                  1.0.1                      py_0    conda-forge
icu                       58.2                 hfc679d8_0    conda-forge
idna                      2.7                   py36_1002    conda-forge
imageio                   2.4.1                      py_0    conda-forge
imagesize                 1.1.0                      py_0    conda-forge
intake                    0.2.7                      py_1    intake
intel-openmp              2019.0                      118  
ipdb                      0.11                       py_1    conda-forge
ipykernel                 5.1.0              pyh24bf2e0_0    conda-forge
ipython                   7.1.1           py36h24bf2e0_1000    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.4.2                      py_0    conda-forge
jasper                    1.900.1              hff1ad4c_5    conda-forge
jedi                      0.13.1                py36_1000    conda-forge
Jinja2                    2.10                      <pip>
jinja2                    2.10                       py_1    conda-forge
jinja2-time               0.2.0                      py_2    conda-forge
jmespath                  0.9.3                      py_1    conda-forge
jpeg                      9c                   h470a237_1    conda-forge
json-c                    0.12.1               h470a237_1    conda-forge
jsonschema                3.0.0a3               py36_1000    conda-forge
jupyter                   1.0.0                      py_1    conda-forge
jupyter_client            5.2.3                      py_1    conda-forge
jupyter_console           6.0.0                      py_0    conda-forge
jupyter_core              4.4.0                      py_0    conda-forge
kealib                    1.4.9                h0bee7d0_2    conda-forge
kiwisolver                1.0.1            py36h2d50403_2    conda-forge
krb5                      1.16.2               hbb41f41_0    conda-forge
libcurl                   7.62.0               hbdb9355_0    conda-forge
libcxx                    7.0.0                h2d50403_2    conda-forge
libdap4                   3.19.1               h18059cb_1    conda-forge
libedit                   3.1.20170329         haf1bffa_1    conda-forge
libffi                    3.2.1                hfc679d8_5    conda-forge
libgdal                   2.2.4                hdb23ba1_9    conda-forge
libgfortran               3.0.1                h93005f0_2  
libiconv                  1.15                 h470a237_3    conda-forge
libkml                    1.3.0                hccc92b1_8    conda-forge
libnetcdf                 4.6.1              h34af652_200    conda-forge
libopenblas               0.3.3                hdc02c5d_3  
libpng                    1.6.35               ha92aebf_2    conda-forge
libpq                     10.5                 hf16a0db_0    conda-forge
libprotobuf               3.6.1                hd28b015_0    conda-forge
libsodium                 1.0.16               h470a237_1    conda-forge
libspatialite             4.3.0a              h3b29d86_23    conda-forge
libssh2                   1.8.0                h5b517e9_2    conda-forge
libtiff                   4.0.9                he6b73bb_2    conda-forge
libtiff                   0.4.2                     <pip>
libuuid                   2.32.1               h470a237_2    conda-forge
libxml2                   2.9.8                h422b904_5    conda-forge
libxslt                   1.1.32               h88dbc4e_2    conda-forge
line_profiler             2.1.2            py36h470a237_1    conda-forge
llvm-meta                 7.0.0                         0    conda-forge
llvmdev                   7.0.0                h2d50403_0    conda-forge
locket                    0.2.0                      py_2    conda-forge
loghub                    0.4.0.dev0                <pip>
lxml                      4.2.5            py36hc9114bc_0    conda-forge
mako                      1.0.7                      py_1    conda-forge
markupsafe                1.1.0            py36h470a237_0    conda-forge
matplotlib                3.0.1                         1    conda-forge
matplotlib-base           3.0.1            py36h45c993b_1    conda-forge
mccabe                    0.6.1                      py_1    conda-forge
memory_profiler           0.54.0                     py_0    conda-forge
metpy                     0.9.1                    py36_0    conda-forge
mistune                   0.8.4            py36h470a237_0    conda-forge
mkl                       2019.0                      118  
mkl_fft                   1.0.6                    py36_0    conda-forge
mkl_random                1.0.2                    py36_0    conda-forge
mock                      2.0.0                 py36_1000    conda-forge
more-itertools            4.3.0                 py36_1000    conda-forge
msgpack-numpy             0.4.4.1                    py_0    conda-forge
msgpack-python            0.5.6            py36h2d50403_3    conda-forge
munch                     2.3.2                      py_0    conda-forge
nbconvert                 5.3.1                      py_1    conda-forge
nbformat                  4.4.0                      py_1    conda-forge
ncurses                   6.1                  hfc679d8_1    conda-forge
netcdf4                   1.4.1           py36ha292673_200    conda-forge
nose                      1.3.7                 py36_1002    conda-forge
notebook                  5.7.0                 py36_1000    conda-forge
numpy                     1.15.4          py36_blas_openblashb06ca3d_0  [blas_openblas]  conda-forge
numpy-base                1.15.4           py36ha711998_0  
numpydoc                  0.8.0                      py_1    conda-forge
olefile                   0.46                       py_0    conda-forge
openblas                  0.3.3                ha44fe06_1    conda-forge
openjpeg                  2.3.0                h316dc23_3    conda-forge
openssl                   1.0.2p               h470a237_1    conda-forge
orderedset                2.0.1                    py36_0    conda-forge
owslib                    0.17.0                     py_0    conda-forge
packaging                 18.0                       py_0    conda-forge
pandas                    0.23.4           py36hf8a1672_0    conda-forge
pandoc                    2.3.1                         0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
param                     1.8.1                      py_0    conda-forge
parse                     1.9.0                      py_0    conda-forge
parse_type                0.4.2                    py36_0    conda-forge
parso                     0.3.1                      py_0    conda-forge
partd                     0.3.9                      py_0    conda-forge
pbr                       4.2.0                      py_0    conda-forge
pcre                      8.41                 hfc679d8_3    conda-forge
pexpect                   4.6.0                 py36_1000    conda-forge
pickleshare               0.7.5                 py36_1000    conda-forge
pillow                    5.3.0            py36hc736899_0    conda-forge
pint                      0.8.1                      py_1    conda-forge
pip                       18.1                  py36_1000    conda-forge
pixman                    0.34.0               h470a237_3    conda-forge
pkgconfig                 1.3.1                 py36_1001    conda-forge
pkginfo                   1.4.2                     <pip>
pluggy                    0.8.0                      py_0    conda-forge
polar2grid                2.3.0a2                   <pip>
pooch                     0.2.0                 py36_1000    conda-forge
poppler                   0.67.0               h4d7e492_3    conda-forge
poppler-data              0.4.9                         0    conda-forge
postgresql                10.5                 ha408888_0    conda-forge
poyo                      0.4.2                      py_0    conda-forge
proj4                     4.9.3                h470a237_8    conda-forge
prometheus_client         0.4.2                      py_0    conda-forge
prompt_toolkit            2.0.7                      py_0    conda-forge
protobuf                  3.6.1            py36hfc679d8_1    conda-forge
psutil                    5.4.3                     <pip>
psutil                    5.4.8            py36h470a237_0    conda-forge
ptyprocess                0.6.0                 py36_1000    conda-forge
py                        1.7.0                      py_0    conda-forge
pycoast                   1.2.0a0.dev0              <pip>
pycodestyle               2.4.0                      py_1    conda-forge
pycountry                 17.9.23                   <pip>
pycparser                 2.19                       py_0    conda-forge
pydecorate                0.1.0                     <pip>
pyepsg                    0.3.2                      py_1    conda-forge
pyflakes                  2.0.0                      py_0    conda-forge
PyGithub                  1.35                      <pip>
pygments                  2.2.0                      py_1    conda-forge
pygrib                    2.0.3            py36h5688137_0    conda-forge
pyhdf                     0.9.10           py36he2eb8c5_0    conda-forge
PyJWT                     1.5.2                     <pip>
pykdtree                  1.3.1                     <pip>
pyopenssl                 18.0.0                py36_1000    conda-forge
pyorbital                 1.3.1                     <pip>
pyparsing                 2.3.0                      py_0    conda-forge
pyproj                    1.9.5.1          py36h508ed2a_6    conda-forge
pyqt                      5.6.0            py36h8210e8a_7    conda-forge
pyresample                1.10.1                   py36_0    conda-forge
pyrsistent                0.14.5           py36h470a237_1    conda-forge
pyshp                     2.0.0                      py_0    conda-forge
pyshp                     1.2.11                    <pip>
pysocks                   1.6.8                 py36_1002    conda-forge
pyspectral                0.8.2                     <pip>
pytest                    3.10.0                py36_1000    conda-forge
pytest-cov                2.6.0                      py_0    conda-forge
python                    3.6.6                h5001a0f_0    conda-forge
python-dateutil           2.7.5                      py_0    conda-forge
python-geotiepoints       1.1.2                     <pip>
python-graphviz           0.8.4                 py36_1002    conda-forge
python-snappy             0.5.3            py36h00d4201_0    conda-forge
python-toggl              0.1.6                     <pip>
pytz                      2018.7                     py_0    conda-forge
pyviz_comms               0.6.0                      py_0    conda-forge
pyyaml                    3.13             py36h470a237_1    conda-forge
pyzmq                     17.1.2           py36hae99301_1    conda-forge
qt                        5.6.2                hd4c90f3_9    conda-forge
qtconsole                 4.4.2                      py_1    conda-forge
rasterio                  1.0.9            py36h1b5fcde_0    conda-forge
ratelim                   0.1.6                    py36_0    conda-forge
readline                  7.0                  haf1bffa_1    conda-forge
requests                  2.20.0                py36_1000    conda-forge
requests-toolbelt         0.8.0                      py_1    conda-forge
ruamel_yaml               0.15.71          py36h470a237_0    conda-forge
s3transfer                0.1.13                py36_1001    conda-forge
satpy                     0.9.1+407.g5297831c           <pip>
scikit-learn              0.20.0          py36_blas_openblash00c3548_201  [blas_openblas]  conda-forge
scipy                     1.1.0           py36_blas_openblashb06ca3d_202  [blas_openblas]  conda-forge
send2trash                1.5.0                      py_0    conda-forge
setuptools                40.5.0                   py36_0    conda-forge
shapely                   1.6.4            py36h164cb2d_1    conda-forge
simplegeneric             0.8.1                      py_1    conda-forge
sip                       4.18.1           py36hfc679d8_0    conda-forge
siphon                    0.8.0                    py36_1    conda-forge
six                       1.11.0                py36_1001    conda-forge
snappy                    1.1.7                hfc679d8_2    conda-forge
snowballstemmer           1.2.1                      py_1    conda-forge
snuggs                    1.4.1                      py_1    conda-forge
sortedcontainers          2.0.5                      py_0    conda-forge
sphinx                    1.8.1                 py36_1000    conda-forge
sphinx-argparse           0.2.3                      py_0    conda-forge
sphinx_copybutton         0.1.3                     <pip>
sphinx_rtd_theme          0.4.2                      py_0    conda-forge
sphinxcontrib-websupport  1.1.0                      py_1    conda-forge
sqlite                    3.25.3               hb1c47c0_0    conda-forge
tblib                     1.3.2                      py_1    conda-forge
terminado                 0.8.1                 py36_1001    conda-forge
testpath                  0.4.2                 py36_1000    conda-forge
tk                        8.6.8                ha92aebf_0    conda-forge
toolz                     0.9.0                      py_1    conda-forge
tornado                   5.1.1            py36h470a237_0    conda-forge
tqdm                      4.28.1                     py_0    conda-forge
traitlets                 4.3.2                 py36_1000    conda-forge
travis-encrypt            0.7.2                 py36_1000    conda-forge
trollimage                1.5.8a0.dev0              <pip>
trollsift                 0.3.1+4.gfed5660           <pip>
twine                     1.11.0                    <pip>
typing                    3.6.4                    py36_0    conda-forge
urllib3                   1.23                  py36_1001    conda-forge
verde                     0.0.1a0+55.ged8ae59           <pip>
versioneer                0.18                       py_1    conda-forge
wcwidth                   0.1.7                      py_1    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.32.2                   py36_0    conda-forge
whichcraft                0.5.2                      py_1    conda-forge
widgetsnbextension        3.4.2                 py36_1000    conda-forge
xarray                    0.10.9+44.gb622c5e7           <pip>
xerces-c                  3.2.0                h5d6a6da_2    conda-forge
xlrd                      1.1.0                      py_2    conda-forge
xz                        5.2.4                h470a237_1    conda-forge
yaml                      0.1.7                h470a237_1    conda-forge
zeromq                    4.2.5                hfc679d8_5    conda-forge
zict                      0.1.3                      py_0    conda-forge
zlib                      1.2.11               h470a237_3    conda-forge

```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

```
</details>

### PR DESCRIPTION
**Pull request by ocefpaf**
Note that, if we don't care about python 2.7 here, we can keep `noarch` and force Python >=3.5 in `host` and `run`. I'm fixing this to accommodate Python 2.7 (PR soon) and I'll leave that decision to the @conda-forge/imageio team.

### CODE REVIEW

**Comment by djhoese** [comment]
Things seem to be working now. Thanks for the quick fix.

**Comment by ocefpaf** [comment]
> Things seem to be working now. Thanks for the quick fix.

No problem. Thanks for the detailed issue that allowed me to easily identify the problem :smile:

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
 source:
-  fn: imageio-{{ version }}.tar.gz
   url: https://pypi.io/packages/source/i/imageio/imageio-{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  sha256: 16b8077bc8a5fa7a58b3e744f7ecbb156d8c088132df31e0f4f546c98de3514a
 
 build:
-  noarch: python
-  script: python -m pip install --no-deps --ignore-installed .
   number: 0
```
Comment (jakirkham) [inline recipe/meta.yaml]: Did we need a build number bump?

**Comment by almarklein** [comment]
Thanks @ocefpaf !

File: recipe/meta.yaml (hunk 1 above)
Comment (almarklein) [reply recipe/meta.yaml]: For the record, not needed because #10 changed the build nummer.

**Pull request merged**